lib/job-iteration/active_record_cursor.rb

```diff
@@ -18,23 +18,16 @@ def initialize
       end
     end
 
-    def initialize(relation, columns = nil, position = nil)
-      @columns = if columns
-        Array(columns)
-      else
-        Array(relation.primary_key).map { |pk| "#{relation.table_name}.#{pk}" }
-      end
+    def initialize(relation, columns, position = nil)
+      @columns = columns
       self.position = Array.wrap(position)
       raise ArgumentError, "Must specify at least one column" if columns.empty?
-      if relation.joins_values.present? && !@columns.all? { |column| column.to_s.include?(".") }
-        raise ArgumentError, "You need to specify fully-qualified columns if you join a table"
```

Comment:
Does it mean that for a multi-table columns the only option is to pass arel attribute instances? I think we may want to keep the `"<table_name>.<column_name>"` as the only accepted format to avoid forcing applications to depend on Arel

Reply:
I reverted this change. The guard clause is back in place now.